api/api.lisp

```diff
@@ -310,6 +312,14 @@
                   ;; specifically eval the rest with the new ram,
                   ;; but NOT the new env
                   (eval-expr-for-p p `(api:begin ,@(cdr rest)) env new-ram))))
+           ((eql api:begin1)
+            (if (null (cdr rest))
+                (eval-expr (car rest) env)
+                (multiple-value-bind (val ignored-env ram)
+                    (eval-expr (car rest) env)
+                  (multiple-value-bind (ignored-val env ram)
+                      (eval-expr-for-p p `(api:begin ,@(cdr rest)) env ram)
+                    (values val env ram)))))
```

Comment:
perhaps moot, but which env should we return?

Reply:
Moot because the returned env is always the same in the Lisp implementation?

I think just returning the original env is probably right, but in case I'm missing something: Is it possible to write a test whose outcome depends on this decision?

Reply:
Here, I am returning the env from the inner begin, which shadows the original env...

Moot, because this env parameter should never be used (the only time I used it, it was a bug). But it seems the invariant is to return the extended env (and then to ignore it).

Reply:
It sounds like you have a handle on it, so just do the thing you think is most consistent, please.

Reply:
I think as it is.